Documentation/Transitions.md

```diff
@@ -0,0 +1,402 @@
+
+![Transition - Present Transition via Segue](https://raw.githubusercontent.com/JakeLin/IBAnimatable-Misc/master/IBAnimatable/PresentTransitionViaSegue.png)
+
+After we select "present portal with dismiss interaction", we need to check the **Module** in Attributes inspector (![Attributes inspector](https://raw.githubusercontent.com/JakeLin/IBAnimatable-Misc/master/IBAnimatable/AttributesInspector.png)). The App may crash if the field is empty. It probably is a bug of Interface Builder. To fix it just simply click on the **Module** field and hit enter, it should show **IBAnimatable**. If everything is ready, we can see the transition animation like this as below . We can have more than one Segue within the same ViewController. They will have different transition animation based on the selected Segue.
+
+![Transition - Present Transition via Segue](https://raw.githubusercontent.com/JakeLin/IBAnimatable-Misc/master/IBAnimatable/PresentTransitionViaSegue.gif)
+
+You can see the example in "TransitionsInInterfaceBuilder.storyboard" and play around different configurations.
+
+## Configuring Push transition in Interface Builder
+
+To use Push transition, we need to have a `UINavigationController`. Firstly, select a `UINavigationController` then configure the constom class to `AnimatableNavigationController` in Identity inspector (![Identity inspector](https://raw.githubusercontent.com/JakeLin/IBAnimatable-Misc/master/IBAnimatable/IdentityInspector.png))
+
+![Transition - Transition using AnimatableNavigationController](https://raw.githubusercontent.com/JakeLin/IBAnimatable-Misc/master/IBAnimatable/TransitionUsingAnimatableNavigationController.png)
+
+Then we can configure **Transition Animation**, **Transition Duration** and **Interactive Gusture** properties in Attributes inspector (![Attributes inspector](https://raw.githubusercontent.com/JakeLin/IBAnimatable-Misc/master/IBAnimatable/AttributesInspector.png)). Those properties are same as in [Configuring Present transition in Interface Builder](#configuring-present-transition-in-interface-builder) section. The difference are we can use `System***Animator`s in **Transition Animation**, e.g. `SystemCube(Left)` for Push transtions.
+
+![Transition - Push Transition settings](https://raw.githubusercontent.com/JakeLin/IBAnimatable-Misc/master/IBAnimatable/PushTransitionSettings.png)
+
+Once we configure those properties, all **Push Transitions** inside the `NavigationController` will use the configured animation effect. By propose, we usually don't have different effects inside the same `NavigationController`, so we don't provide Push Segues to support different transition animations. If you still want to have different transition animations, we can use Apple default API to achieve that. Please check out `UINavigationControllerDelegate` and `UIViewControllerTransitioningDelegate`.
```

Comment:
@tbaranes  Removed `By propose`

We do support changing the `AnimationType` property in `AnimatableNavigationController`. But we will have weird issues like the all transition animation will be changed within the same `AnimatableNavigationController`. Most of the time, we would have only one transition animation within one NavigationController. In that case, we can configure it in Interface Builder. If the developer still wants to have a different one, I still recommend using `UINavigationControllerDelegate` to do that. The developer needs to handle potential weird issues I mentioned before. But this issue is not caused by `IBAnimatable` at all. It maybe an Apple architecture issue.